spec/features/trusted_login_flow_spec.rb

```diff
@@ -0,0 +1,131 @@
+require 'rails_helper'
+
+feature 'Sign in using trusted parameters', js: true do
+  background do
+    load 'db/seeds.rb'
+    create_default_application
+  end
+  let(:role) { 'instructor' }
+  let(:payload) {
+    {
+      role:  role,
+      external_user_uuid: SecureRandom.uuid,
+      name:  'Tester McTesterson',
+      email: 'test@test.com',
+      school: 'Testing U'
+    }
+  }
+  let(:signed_params) {
+    { sp: OpenStax::Api::Params.sign(params: payload, secret: @app.secret) }
+  }
+
+  describe 'instructors' do
+    it 'signs in and links' do
+      user = create_user 'user'
+      arrive_from_app(params: signed_params)
+      expect_sign_in_page
+      complete_login_username_or_email_screen 'user'
+      complete_login_password_screen 'password'
+      expect_back_at_app
+      expect_validated_records(params: payload, user: user)
+    end
+
+    it 'can sign up with trusted data' do
+      arrive_from_app(params: signed_params)
+      expect_sign_in_page
+      click_sign_up
+      expect_sign_up_page
+
+      expect(page).not_to have_field('signup_role') # no changing the role
+      expect(page).to have_field('signup_email', with: payload[:email])
+      click_button(t :"signup.start.next")
+      wait_for_animations
+      click_button(t :"signup.start.next")
+      complete_signup_password_screen('password')
+      expect_signup_profile_screen
+      expect(page).to have_field('profile_first_name', with: 'Tester')
+      expect(page).to have_field('profile_last_name', with: 'McTesterson')
+      expect(page).to have_field('profile_school', with: payload[:school])
+      complete_signup_profile_screen_with_whatever
+
+      expect_back_at_app # note, no "verification pending" step
+      expect_validated_records(params: payload)
+    end
+
+    it 'requires email validation when modified' do
+      arrive_from_app(params: signed_params)
+      expect_sign_in_page
+      click_sign_up
+      expect_sign_up_page
+
+      email = 'test-modified-teacher@test.com'
+
+      fill_in (t :"signup.start.email_placeholder"), with: email
+      click_button(t :"signup.start.next")
+      wait_for_animations
+      click_button(t :"signup.start.next")
+      expect_signup_verify_screen
+      ss = SignupState.find_by!(contact_info_value: email)
+      fill_in (t :"signup.verify_email.pin"), with: ss.confirmation_pin
+      click_button(t :"signup.verify_email.confirm")
+      complete_signup_password_screen('password')
+      expect_signup_profile_screen
+      complete_signup_profile_screen_with_whatever(role: :instructor)
+      expect_back_at_app
+      expect_validated_records(params: payload.merge(email: email))
+    end
+
+  end
+
+  describe 'students' do
+    let(:role) { 'student' }
+
+    it 'signs in and links' do
```

Comment:
think this title is wrong - this is sign up, probably just copy/paste from above

Reply:
yeah probably copy & paste, or maybe I meant that the user is signed in at the end.  Either way it's not well named 😁